SonarQube.Client/Requests/AggregatingRequestFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarQube.Client.Requests
+{
+    internal class AggregatingRequestFactory : IRequestFactory
+    {
+        private readonly IRequestFactory[] requestFactories;
+
+        public AggregatingRequestFactory(params IRequestFactory[] requestFactories)
+        {
+            this.requestFactories = requestFactories;
+        }
+
+        public TRequest Create<TRequest>(ServerInfo serverInfo) where TRequest : class, IRequest
+        {
+            foreach (var factory in requestFactories)
+            {
+                var newRequest = factory.Create<TRequest>(serverInfo);
+                if (newRequest != null)
```

Comment:
The main reason I added the `class` constraint was so this comparison is against `null` rather than `default(T)`.
We know the instances are all going to be classes, so it seemed clearer to make it explicit.